setup.py

```diff
@@ -41,7 +41,7 @@ def find_version(*paths):
         "pyarrow == 10.0.1",
         "pydantic == 1.10.4",
         "ray[default] ~= 2.0",
-        "s3fs == 2022.2.0",
```

Comment:
`pyiceberg 0.3.0` requires `2023.1.0` in its specification. The version range will be widen in the next release (0.4.0)